src/merge.py

```diff
@@ -0,0 +1,54 @@
+"""Implementation of merge sort in Python."""
+
+
+def merge_sort(iter):
+    """Sort the iterable using the merge sort method."""
+    if not isinstance(iter, (list, tuple)):
+        raise TypeError("Input only a list/tuple of integers")
+    if not all(isinstance(x, (int, float)) for x in iter):
+        raise ValueError("Input only a list/tuple of integers")
+
+    if len(iter) == 1:
+        return iter
+    if len(iter) == 2:
+        if iter[0] > iter[1]:
+            tmp = iter[0]
+            iter[0] = iter[1]
+            iter[1] = tmp
```

Comment:
These 3 lines could be accomplished in one line and would eliminate the need for that intermediate `tmp` variable